src/utils/createElement.js

```diff
@@ -0,0 +1,68 @@
+export function createSpan({ className, txt }) {
+    const newSpan = document.createElement("span");
+    if (className) newSpan.setAttribute("class", className);
+    if (txt) newSpan.innerHTML = txt;
+
+    return newSpan;
+}
+
+export function createBtn({ className, txt }) {
+    const newBtn = document.createElement("button");
+    if (className) newBtn.setAttribute("class", className);
+
+    return newBtn;
+}
+
+export function createDiv({ className, txt }) {
+    const newDiv = document.createElement("div");
+    if (className) newDiv.setAttribute("class", className);
+    if (txt) newDiv.innerHTML = txt;
+
+    return newDiv;
+}
+
+export function createNav({ className, txt }) {
+    const newNav = document.createElement("nav");
+    if (className) newNav.setAttribute("class", className);
+
+    return newNav;
+}
```

Comment:
```suggestion
// 1
function newElement(tagName, txt, attributes) {
    const $element = document.createElement(tagName);
    if (txt) $element.innerHTML = txt;
    for(let key in attributes) {
        $element.setAttribute(key, attributes[key])
    }
    return $element;
}

newElement('a', '링크 태그입니다.', { class: '클래스입니다', href: '/#' })
// Result: <a class="클래스입니다" href="/#">링크 태그입니다.</a>

// 2
const create = {}
const block = ['span', 'button', 'div', 'nav', 'ul', 'li' ]
block.forEach((element) => {
    create[element] = (className, txt) => {
        const newElement = document.createElement(element)
        if (className) newElement.setAttribute("class", className);
        if (txt) newElement.innerHTML = txt;
        return newElement
    };
}, {});


create.div("클래스입니다", "div 태그입니다.");
// Result : <div class=​"클래스입니다">​div 태그입니다.​</div>​
```

중복된 코드가 많은 것 같아 이를 줄이는 방법에 대해서 많이 고민해보았는데, 이런 방식으로 여러 함수를 한번에 정의 하는 방법은 어떨지 제안해봅니다! :)

Reply:
감사합니다!! 리팩토링할 때 참고해서 수정하겠습니다😊